internals/overlord/servstate/manager_test.go

```diff
@@ -2146,10 +1950,21 @@ func (f fakeLogManager) ServiceStarted(service *plan.Service, logs *servicelog.R
 	// no-op
 }
 
-func (s *S) insertDoneCheck(c *C, service string) string {
-	return fmt.Sprintf("sync; touch %s", filepath.Join(s.dir, service))
+// insertDoneChecks modifies layer service commands which contains a
+// $NOTIFY_DONE_CHECK barrier placeholder. The placeholder is replaced
+// with a command which writes a service specific file to a test
+// directory, allowing waitForDoneCheck to detect service side effect
+// completion.
+func (s *S) insertDoneChecks(c *C, layer *plan.Layer) {
+	for _, service := range layer.Services {
+		doneCheck := fmt.Sprintf("sync; touch %s", filepath.Join(s.dir, service.Name))
+		service.Command = strings.Replace(service.Command, "$NOTIFY_DONE_CHECK", doneCheck, -1)
```

Comment:
I don't love this, as it looks like an environment variable, whereas it's really a template variable substitution. Can we just explicitly add it to the list of fmt.Sprintf arguments (most of the layers seem to be built using Sprintf anyway), like `fmt.Sprintf("... %s ...", s.doneCheck("svc1"))` or similar?

Alternatively, if you want to stick with the current approach, let's use the string `{{.NotifyDoneCheck}}` as we've done elsewhere in Pebble with `{{.ProgramName}}` (Go template syntax -- though we don't have to use a full Go template).

Reply:
```{{.NotifyDoneCheck}}``` I've adopted this.

The reason for not using %s (deliberately removing this) was because the string snippets requiring the checks typically require insertion in more than one place, and also includes other %s fields as well. The end result was long Sprintf statements, which was difficult to follow.

The actual test code, in my opinion, looks much cleaner now - I think :) But let me know if this feels OK now. If you prefer the previous approach, no problem at all - happy to revert.

Reply:
Let me know how you feel about this please :) ?

Reply:
Yep, this is good, thanks.